src/create-glamorous.js

```diff
@@ -42,26 +45,29 @@ function createGlamorous(splitProps) {
        */
       const GlamorousComponent = withTheme(
         (props, context) => {
-          /* eslint no-use-before-define: 0 */
+          const updateClassName = shouldUpdate(props, context)
+
           const {toForward, cssOverrides, cssProp} = splitProps(
             props,
             GlamorousComponent,
           )
 
           // create className to apply
-          const className = getGlamorClassName({
-            styles: GlamorousComponent.styles,
-            props,
-            cssOverrides,
-            cssProp,
-            context,
-            displayName: GlamorousComponent.displayName,
-          })
+          GlamorousComponent.className = updateClassName ?
```

Comment:
I believe GlamorousComponent would have to be converted from a functional component to a class... would you be okay with that? We could also potentially _only_ make a stateful component if `shouldClassNameUpdate` is set.